core/src/smartcontracts/wasm.rs

```diff
@@ -1241,28 +1392,44 @@ impl<'wrld> import_traits::ExecuteOperations<state::executor::ValidateQuery<'wrl
     #[codec::wrap]
     fn execute_query(
         query_request: SmartContractQueryRequest,
-        state: &state::executor::ValidateQuery<'wrld>,
+        state: &mut state::executor::ValidateQuery<'wrld>,
```

Comment:
seems like reason to not make query store part of the wsv

Reply:
i think this happened not because of query store, since it uses only `&self` methods, but because we now attach `HashSet<QueryId>` to `State` and we need to update it.